microprofile/WFLY-12609_MicroProfile_JWT_1_1.adoc

```diff
@@ -0,0 +1,109 @@
+
+This integration will not be supporting the use of pre-configured SecurityDomains, the general motivation to use MicroProfile JWT is to use an authentication mechanism where identities are self described using the tokens exchanged eliminating the need for configured security resources for the secured endpoint(s).  Use cases could be identified where pre-configured resources add an additional benefit but those would be reviewed under their own enhancements.
+
+It is assumed when working with the MicroProfile specifications that we will be supporting a single deployment, this deployment may contain sub-deployments.  The SecurityDomain of each top level deployment will be isolated from the SecurityDomain of any parallel top level deployment.  Advanced WildFly Elytron features such identity propagation across domains / top level deployments will not be supported.  At a later point we could consider an enhancement to offer more control of the SecurityDomains in use but that is out of scope for this enhancement.
+
+== Implementation Plan
+
+The primary entry point for this enhancement will be a new subsystem `microprofile-jwt` which will be added to the WildFly project.  The WildFly project will also pull in SmallRye JWT as well as the MicroProfile JWT API.
+
+Finally there will be an additional project `Elytron JWT` added to the WildFly Elytron project, this will also depend on SmallRye JWT, the purpose of this project will be to add any tighter integration required with the WildFly Elytron APIs.  The contents of the `Elytron JWT` project will be considered private API and will be subject to change between releases.
+
+If enhancements are required these will be submitted directly to the SmallRye JWT project, the Elytron JWT project will be a back up if we need a location to hold enhancements whilst the availability in SmallRye JWT is considered.
+
+== Test Plan
+
+Details are still to be discussed but in general testing will make use of the MicroProfile TCK, additionally integration tests will be added to the WildFly testsuite using deployments configured to use MicroProfile JWT.  Any code added to WildFly Elytron will also be tested using co-located unit tests.
+
+One aspect we may consider is the generation of tokens as that is outside the specification, within the testsuites and possibly as a stand alone example we may want an endpoint which is capable of generating the JWT tokens - this would not be a part of our supported distribution however it would be useful to allow for deployments to use MicroProfile JWT without additional infrastructure requirements.  Something such as this should have the ability to take a WildFly Elytron SecurityIdentity and convert it into a JWT token.
+
+== Community Documentation
+
+Documentation will be added to the WildFly documentation showing how a MicroProfile JWT deployment can be deployed, we may want to restrict the level of detail as the behaviour will predominantly be covered by the specification.
+
+
```

Comment:
What are the plans for Quickstart ? Have you spoke with @xstefank and @emmartins ?

Reply:
Will review options for Quickstarts later and update here as required.

Reply:
The quickstart is now tracked separately.

Reply:
@darranl Do you plan to reflect current state regarding quickstarts in this paragraph?

Reply:
No the Quickstart will be handled under it's own RFE at the moment so that will have it's own analysis.

Reply:
@darranl Sorry, seems like I checked older revision. This is OK.

Reply:
@darranl My bad again, but when I was writing previous comment, only top paragraph was visible for me for some reason. What I;m talking about is this:

> ...quickstarts should be considered to ensure a quickstart exists that can be used with WildFly.

As you said, they've already been considered and are going to be added as a separate feature.

Reply:
Please replace ^^^ by something like "quickstart will be processed as a separate proposal".